### PR TITLE
Add Dockerfile (part 2)

### PR DESCRIPTION
**Pull request by turt2live**
Continuation of https://github.com/vector-im/riot-web/pull/7771
Fixes https://github.com/vector-im/riot-web/issues/315

I've opened a new PR for this instead of committing directly to the author's original branch because their branch is named "develop" and I don't want to accidentally break anything.

Changes from the original concept:
* Reduce build context size for dirty environments (developers)
* Support using custom SDKs - this is to support us having a `:develop` or similar image if we want it
* Use `yarn` now that we've switched to it
* Misc improvements (layer reduction, docs, etc)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -259,6 +259,42 @@ To change the config.json for the desktop app, create a config file which will b
...
+To supply your own custom `config.json`, map a volume to `/app/config.json`. For example, 
+if your custom config was located at `/etc/riot-web/config.json` then your Docker command
+would be:
+```bash
+docker run -p 80:80 -v /etc/riot-web/config.json:/app/config.json vectorim/riot-web
+```
+
+To build the image yourself:
+```bash
+git clone https://github.com/vector-im/riot-web.git riot-web
```
Comment (jryans) [inline README.md]: The checkout directory will default to `riot-web`, so the last argument here isn't needed.

File: README.md [hunk 2]
```diff
@@ -259,6 +259,42 @@ To change the config.json for the desktop app, create a config file which will b
 
 In the paths above, `$NAME` is typically `Riot`, unless you use `--profile $PROFILE` in which case it becomes `Riot-$PROFILE`.
 
+Running from Docker
+===================
+
+The Docker image can be used to serve riot-web as a web server. The easiest way to use 
+it is to use the prebuilt image:
+```bash
+docker run -p 80:80 vectorim/riot-web
```
Comment (jryans) [inline README.md]: Is `vectorim` an existing account on Docker Hub already? It seems like `matrixdotorg` does exist, but `vectorim` does not... I agree it likely makes sense for Riot to be in a separate `vectorim`-like account, but we should create it first before publishing these docs if it hasn't been done yet. (Also, would `newvector` or something be better...?)

It's also quite possible `vectorim` already does exist and I just don't understand Docker Hub at all. 😅

Comment (turt2live) [reply README.md]: I'm the best at including relevant details and omitted the plan here, sorry. The plan is to nag people internally for opinions first and then merge this

**Comment by turt2live** [comment]
> How would publishing to Docker Hub work? Will we add that to our release steps for official releases, or...?

It'll magically work, once I set it up. Docker hub does builds for us (although they are painfully slow) and can be matched to branches and such. At worst we'll have to make sure that the riot-web push is the last one of the 3 repos or it'll end up building the wrong version.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ARG REACT_SDK_BRANCH="master"
+ARG JS_SDK_REPO="https://github.com/matrix-org/matrix-js-sdk.git"
+ARG JS_SDK_BRANCH="master"
+
+RUN apk add --no-cache git dos2unix
+
+WORKDIR /src
+
+COPY . /src
+RUN dos2unix /src/scripts/docker-link-repos.sh && sh /src/scripts/docker-link-repos.sh
```
Comment (SISheogorath) [inline Dockerfile]: It makes no sense to convert it during the docker build. This should be done before adding it to the repository.

Comment (turt2live) [reply Dockerfile]: It is committed with the right line endings, but Windows tends to not care once it is on disk.

Comment (SISheogorath) [reply Dockerfile]: In this case it's not Windows but git on Windows that has automatic line conversion enabled. This should be fixable by telling git to not modify them:

```
# .gitattributes
*.sh		text eol=lf
```

For details see: https://git-scm.com/docs/gitattributes

Comment (turt2live) [reply Dockerfile]: I've tried everything in the book, trust me.

**Comment by turt2live** [comment]
Merging in hopes that it triggers a build

**Pull request merged**